### ISSUE TITLE
Rule Request: Ignore CodingKey enums in Nesting

### ISSUE THREAD

**Issue by braker1nine** [realm/SwiftLint#5641]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### New rule request

This is more of an edit to an existing rule than a new rule, so apologies if this is the wrong format... Add a boolean option to the Nesting rule. When this option is enabled, enum's conforming to `CodingKey` won't count as violations of nesting.

My main argument for this is that CodingKey's are less of a nested "type" and more of an annotation of the current type for `Codable` If you have types that have custom `CodingKey` types, you basically lose a 

**Triggering**
```swift
struct Wrapper: Codable {
    struct Inner: Codable {
        struct WayInner: Codable {
        }
    }
}
```

```swift
struct Wrapper: Codable {
    enum CodingKeys: String, CodingKey {
        case id = "identifier"

        struct TechnicallyOkayButBad {
            struct Violation {}
         }
    }
}
```

**Non Triggering**
```swift
struct Wrapper: Codable {
    struct Inner: Codable {
        let id: Int
        
        enum CodingKeys: String, CodingKey {
            case id = "identifier
        }
    }
}
```

I think I would propose this be an **opt-in setting** on the nesting rule.

 Looking at the criteria in the README
> A  rule that can have many false positives

I think this change would actually reduce false positives

> A rule that is too slow

I don't _think_ this would be an issue

> A rule that is not general consensus or is only useful in some cases

 I'm not sure if there's consensus here. And it's a very specific case...

**Comment by braker1nine** [comment]
I've already spent some time on a rough version of implementation. So if this is something that feels worth adding, I'm happy to work in a PR for it

**Comment by SimplyDanny** [comment]
Sounds like a reasonable exclusion. However, I recommend adding this only as an option to the rule. So the default behavior of the rule doesn't change.

If you have worked on a PR already, please go ahead an propose it. I'll be glad to have a look at it.
